### DISCUSSION TITLE
ESLint show one error at the time when using ecmaVersion: 5

### DISCUSSION THREAD

**Discussion by jcubic** [eslint/eslint#18684] [Q&A] [answered]
I've used default config using `npm init` like in docs and add `ecmaVersion: 5` (I have ES5 project that I need to lint, so I don't add new features by accident).

```javascript
import globals from "globals";
import pluginJs from "@eslint/js";


export default [
  {
    files: ["**/*.js"],
    languageOptions: {
      ecmaVersion: 5,
      sourceType: "script"
    }
  },
  {
    languageOptions: {
      globals: globals.browser
    }
  },
  pluginJs.configs.recommended
];
```

And when I run `npx eslint` I got single error, when I fix the error I get next error. In different project I always get all the errors at once.

If I remove `ecmaVersion` version, I got different errors, but they all appear at the same time.

This is my project https://github.com/jcubic/sysend.js/tree/eslint

For me, it looks like a bug, but I first wanted to ask in discussions, maybe it's a known issue.

**EDIT**: When I fixed all ES5 errors, I got all the same errors, same as without version.

**Comment by aladdin-add** [comment, marked as answer]
I see that you are using some ES6 features in your code, such as object destructuring and rest parameters. When setting the ecmaVersion to ES5, you'll encounter parsing errors (`Parsing error: Unexpected token`), and ESLint rules will not be applied.

**Reply by jcubic** [reply]
Yes, I'm in a process of removing them. Ok, thanks. That make sense.
